src/app/shared/services/media-state.service.ts

```diff
@@ -1,33 +1,76 @@
-  private readonly mediaSubject: Map<string, BehaviorSubject<Media>> = new Map()
+export class MediaStateService implements OnDestroy {
+  private readonly mediaSubjects: Map<string, BehaviorSubject<Media | undefined>> = new Map()
+  private readonly subscriptions: EventSubscription[] = []
+  private readonly logger: Logger
 
-  constructor(private readonly mediaService: MediaService) {}
+  constructor(
+    private readonly mediaService: MediaService,
+    private readonly mediaEventObserver: MediaEventObserver,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('MediaStateService')
+    lastValueFrom(this.mediaService.getAllMedia())
+      .then(mediaAssets => mediaAssets.forEach(media => this.updateMedia(media)))
+      .catch(error => this.logger.data(error).error(`Failed while fetching all media from server`))
+    this.subscribeToEvents()
+  }
 
-  public async subscribeToMedia(id: string): Promise<Observable<Media>> {
-    const mediaSubject: BehaviorSubject<Media> = await this.createMediaSubject(id)
-    return mediaSubject.asObservable()
+  private updateMedia(media: Media): void {
+    const mediaSubject: BehaviorSubject<Media | undefined> | undefined = this.getMediaSubject(media.sourceName)
+    mediaSubject.next(media)
   }
 
-  private async createMediaSubject(id: string): Promise<BehaviorSubject<Media>> {
-    const mediaSubject: BehaviorSubject<Media> | undefined = this.mediaSubject.get(id)
-    if (mediaSubject) {
-      return mediaSubject
+  private getMediaSubject(sourceName: string): BehaviorSubject<Media | undefined> {
+    const mediaSubjectKey: string = this.getMediaSubjectKey(sourceName)
+    if (!this.mediaSubjects.has(mediaSubjectKey)) {
+      this.mediaSubjects.set(mediaSubjectKey, new BehaviorSubject<Media | undefined>(undefined))
     }
-    return await this.getCleanMediaSubject(id)
+    return this.mediaSubjects.get(mediaSubjectKey)!
+  }
+
+  private getMediaSubjectKey(sourceName: string): string {
+    return sourceName.toUpperCase()
+  }
+
+  private subscribeToEvents(): void {
+    this.subscriptions.push(this.mediaEventObserver.subscribeToMediaCreated((event: MediaCreatedEvent) => this.updateMedia(event.media)))
+    this.subscriptions.push(this.mediaEventObserver.subscribeToMediaDeleted((event: MediaDeletedEvent) => this.deleteMediaFromSubjects(event)))
   }
 
-  private async getCleanMediaSubject(id: string): Promise<BehaviorSubject<Media>> {
-    const media: Media = await this.fetchMedia(id)
-    return new BehaviorSubject<Media>(media)
+  private deleteMediaFromSubjects(event: MediaDeletedEvent): void {
+    this.mediaSubjects.forEach(subject => {
+      if (subject.value?.id === event.mediaId) {
+        subject.next(undefined)
+      }
+    })
+  }
+
+  public subscribeToMedia(sourceName: string): Observable<Media | undefined> {
+    return this.createMediaSubject(sourceName).asObservable()
+  }
+
+  private createMediaSubject(sourceName: string): BehaviorSubject<Media | undefined> {
```

Comment:
You no longer need `createMediaSubject()`. 
`getMediaSubject()` does the exact same as `createMediaSubject()` so one of them should be deleted.

Reply:
A very good point, i removed create!